packages/devtools_app/lib/src/scaffold.dart

```diff
@@ -274,14 +274,17 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
     });
 
     // If we had no explicit page, we want to write one into the URL but
-    // without triggering a navigation.
-    if (widget.page == null) {
-      final routerDelegate = DevToolsRouterDelegate.of(context);
-      Router.neglect(context, () {
-        routerDelegate.navigateIfNotCurrent(
-          _currentScreen.screenId,
-          routerDelegate.currentConfiguration.args,
-        );
+    // without triggering a navigation. Since we can't nagivate during a build
+    // we have to wrap this in `Future.microtask`.
+    if (widget.page == null && _currentScreen is! SimpleScreen) {
+      Future.microtask(() {
+        final routerDelegate = DevToolsRouterDelegate.of(context);
+        Router.neglect(context, () {
+          routerDelegate.navigateIfNotCurrent(
```

Comment:
Instead of using Future.microtask(), what about WidgetsBinding.instance.addPostFrameCallback ?

another idea - is there a better way to update the url that doesn't trigger a build in the first place?

Reply:
I think it is forcing build to be called because `routerDelegate.navigateIfNotCurrent` eventually calls `notifyListeners`. One idea is to hide the `notifyListeners` call behind a bool `if (shouldNotify)`. `shouldNotify` could be an optional named parameter that defaults to true. I haven't tried this locally, so we'll need to make sure that this doesn't break any routing behavior if `notifyListeners` was false. This may cause more trouble than it is worth, but could be worth looking into.

Reply:
My understanding is that `notifyListeners()` is how we notify Flutter that the URL may have changed (and in turn, it may update the visible URL to the user). If we remove that, I think the visible URL in the browser won't change at all (since we've only updated our own local state variables).

(If this doesn't seem right, I can do some testing of it though)

Reply:
That makes sense. The postFrameCallback seems fine then.